java/core/src/main/java/com/suse/manager/reactor/messaging/JobReturnEventMessageAction.java

```diff
@@ -145,103 +145,12 @@ public void execute(EventMessage msg) {
-                throw e;
-            }
-
-            saltServerActionService.handleActionChainResult(jobReturnEvent.getMinionId(),
-                    jobReturnEvent.getJobId(),
-                    actionChainResult,
-                    stateResult -> false);
-
-            if (isFunctionTestMode) {
-                return;
-            }
-
-            Set<PackageChangeOutcome> packageRefreshNeeded = actionChainResult.entrySet().stream()
-                    .map(entry -> SaltActionChainGeneratorService.parseActionChainStateId(entry.getKey())
-                            .map(stateId -> handlePackageChanges(jobReturnEvent, entry.getValue().getName(),
-                                    Optional.ofNullable(entry.getValue().getChanges().getRet())))
-                            .orElse(PackageChangeOutcome.DONE)
-                    ).collect(Collectors.toSet());
-
-            if (packageRefreshNeeded.contains(PackageChangeOutcome.NEEDS_DELAYED_REFRESHING)) {
-                schedulePackageRefresh(scheduler.get(), jobReturnEvent.getMinionId(),
-                                       Date.from(Instant.now().plusSeconds(60)));
-            }
-            else if (packageRefreshNeeded.contains(PackageChangeOutcome.NEEDS_REFRESHING)) {
-                schedulePackageRefresh(scheduler.get(), jobReturnEvent.getMinionId());
-            }
+            handleEventInActionChain(jobResult, jobReturnEvent, isFunctionTestMode);
         });
 
         //For all jobs except when action chains are involved or the action was in test mode
         if (!isActionChainInvolved && !isFunctionTestMode) {
-            Date earliest = new Date();
-            PackageChangeOutcome changeOutcome = handlePackageChanges(jobReturnEvent,
-                            Optional.ofNullable(function).map(Xor::right), jobResult);
-            if (changeOutcome != PackageChangeOutcome.DONE) {
-                if (changeOutcome == PackageChangeOutcome.NEEDS_DELAYED_REFRESHING) {
-                    earliest = Date.from(Instant.now().plusSeconds(60));
-                }
-                Optional<User> scheduler = Optional.empty();
-                if (actionId.isPresent()) {
-                    Optional<Action> action = Optional.ofNullable(ActionFactory.lookupById(actionId.get()));
-                    if (action.isPresent()) {
-                        scheduler = Optional.ofNullable(action.get().getSchedulerUser());
-                        if (action.get().getActionType().equals(ActionFactory.TYPE_DIST_UPGRADE)) {
-                            Calendar calendar = Calendar.getInstance();
-                            calendar.add(Calendar.SECOND, 30);
-                            earliest = calendar.getTime();
-                        }
-                    }
-                }
-                schedulePackageRefresh(scheduler, jobReturnEvent.getMinionId(), earliest);
-            }
+            handleNotActionChainNorTestMode(jobResult, jobReturnEvent, function, actionId);
```

Comment:
I would name the method `handleStandaloneAction()`. To be even more clear you could do:

```java
        //For all jobs except when action chains are involved or the action was in test mode
        boolean isStandaloneAction = !isActionChainInvolved && !isFunctionTestMode;
        if (isStandaloneAction) {
            handleStandaloneAction(jobResult, jobReturnEvent, function, actionId);
        }
```